internal/pkg/manifest/validate.go

```diff
@@ -1412,10 +1411,35 @@ func (p PublishConfig) validate() error {
 
 // validate returns nil if Topic is configured correctly.
 func (t Topic) validate() error {
-	if t.Type != nil && !contains(aws.StringValue(t.Type), validTopicsTypeValues) {
-		return fmt.Errorf(`"type" value %q is not allowed; must be one of %s`, aws.StringValue(t.Type), english.WordSeries(validTopicsTypeValues, "or"))
+	if err := validatePubSubName(aws.StringValue(t.Name)); err != nil {
+		return err
+	}
+	if err := t.Fifo.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validate returns nil if FifoAdvanceConfigOrBool is configured correctly.
+func (f FifoAdvanceConfigOrBool) validate() error {
+	if f.IsEmpty() {
+		return nil
 	}
-	return validatePubSubName(aws.StringValue(t.Name))
+	if f.Enable != nil {
+		return nil
+	}
+	if err := f.Advanced.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validate returns nil if FifoAdvanceConfig is configured correctly.
+func (a FifoAdvanceConfig) validate() error {
+	if a.IsEmpty() {
+		return fmt.Errorf("content_based_deduplication is not defined correctly")
+	}
+	return nil
```

Comment:
what do you think of 
```suggestion
        if f.IsEmpty() {
                return nil
        }
	return f.Advanced.validate()
}

// validate returns nil if FifoAdvanceConfig is configured correctly.
func (a FifoAdvanceConfig) validate() error {
	return nil
}
```

My concern is that, with the current implementation, `func (a FifoAdvanceConfig) validate() error` can only be invoked when `!f.IsEmpty()` and when `f.Enabled == nil` - otherwise there would be validation bug. This introduces implicit tangling between the caller and callee and could be error-prone in the future.